include/OpenSpaceToolkit/Astrodynamics/Trajectory/StateBuilder.hpp

```diff
@@ -0,0 +1,142 @@
+
+namespace ostk
+{
+namespace astro
+{
+namespace trajectory
+{
+
+using ostk::core::types::Shared;
+using ostk::core::ctnr::Array;
+
+using ostk::math::obj::VectorXd;
+
+using ostk::physics::coord::Frame;
+
+using ostk::physics::time::Instant;
+
+using ostk::astro::trajectory::State;
+using ostk::astro::trajectory::state::CoordinatesBroker;
+using ostk::astro::trajectory::state::CoordinatesSubset;
+
+/// @brief                      Factory class to generate States with common reference frame and coordinates subsets
+
+class StateBuilder
+{
+   public:
+    /// @brief                  Constructor.
+    ///
+    /// @param                  [in] aFrameSPtr The reference frame in which the coordinates are referenced to and
+    /// resolved in
+    /// @param                  [in] aCoordinatesSubsetsArray The array of coordinates subsets defining the output
+    /// States
+
+    StateBuilder(
+        const Shared<const Frame>& aFrameSPtr, const Array<Shared<const CoordinatesSubset>>& aCoordinatesSubsetsArray
+    );
+
+    /// @brief                  Constructor.
+    ///
+    /// @param                  [in] aFrameSPtr The reference frame in which the coordinates are referenced to and
+    /// resolved in
+    /// @param                  [in] aCoordinatesBroker Shared pointer to an existing Coordinates Broker
+
+    StateBuilder(const Shared<const Frame>& aFrameSPtr, const Shared<const CoordinatesBroker>& aCoordinatesBrokerSPtr);
+
+    /// @brief                  Equality operator.
+    ///
+    /// @param                  [in] aStateBuilder The StateBuilder to compare to
+    /// @return                 True if the  StateBuilders are equal, false otherwise
+
+    bool operator==(const StateBuilder& aStateBuilder) const;
+
+    /// @brief                  Inequality operator.
+    ///
+    /// @param                  [in] aStateBuilder The StateBuilder to compare to
+    /// @return                 True if the  StateBuilders are not equal, false otherwise
+
+    bool operator!=(const StateBuilder& aStateBuilder) const;
+
+    /// @brief                  Stream insertion operator.
+    ///
+    /// @param                  [in] anOutputStream The output stream to insert into
+    /// @param                  [in] aStateBuilder The StateBuilder to insert
+    /// @return                 The output stream with the StateBuilder inserted
+
+    friend std::ostream& operator<<(std::ostream& anOutputStream, const StateBuilder& aStateBuilder);
+
+    /// @brief                  Check if the StateBuilder is defined.
+    ///
+    /// @return                 True if the StateBuilder is defined, false otherwise
+
+    bool isDefined() const;
+
+    /// @brief                  Produce a State linked to the Frame and Coordinates Broker of the StateBuilder.
+    ///
+    /// @return                 A State linked to the Frame and Coordinates Broker of the StateBuilder
+
+    const State buildState(const Instant& anInstant, const VectorXd& aCoordinates) const;
```

Comment:
```suggestion
    const State build(const Instant& anInstant, const VectorXd& aCoordinates) const;
```
Nit: since it's called StateBuilder, it's maybe okay to leave out the `State` as it's known what it will be building.

Reply:
Yeah I do agree here. That puts us at:
`StateBuilder::build` for C++ and
`StateBuilder.state` for Python

If we're going to be brief with the names, should probably be consistent.

Reply:
➕ on just `.build` for both languages